--- a/ufl_legacy/measure.py
+++ b/ufl_legacy/measure.py
@@ -65,7 +65,7 @@
 integral_type_to_measure_name = dict((l, s) for l, s in _integral_types)
 measure_name_to_integral_type = dict((s, l) for l, s in _integral_types)
 
-custom_integral_types = ("custom", "cutcell", "interface", "overlap")
+custom_integral_types = ("custom", "cutcell", "interface", "overlap", "exterior_cut_facet")
 point_integral_types = ("vertex",)  # "point")
 facet_integral_types = ("exterior_facet", "interior_facet")
 
